presto-main/src/main/java/com/facebook/presto/memory/ClusterMemoryManager.java

```diff
@@ -52,6 +93,87 @@ public void process(Iterable<QueryExecution> queries)
             }
         }
         clusterMemoryUsageBytes.set(totalBytes);
+
+        updateNodes();
+        updatePools();
+    }
+
+    @VisibleForTesting
+    synchronized Map<MemoryPoolId, ClusterMemoryPool> getPools()
+    {
+        return ImmutableMap.copyOf(pools);
+    }
+
+    private void updateNodes()
```

Comment:
so you only check of memory once?  I expected to see a scheduler checking every N-seconds